forge-core/src/main/java/forge/deck/DeckRecognizer.java

```diff
@@ -760,6 +762,9 @@ private DeckSection getTokenSection(String deckSec, DeckSection currentDeckSecti
         // is not supported, but other possibilities exist (e.g. Commander card in Constructed
         // could potentially go in Main)
         DeckSection matchedSection = DeckSection.matchingSection(card);
+        // If it's a commander candidate, put it there.
+        if (matchedSection == DeckSection.Main && this.isAllowed(DeckSection.Commander) && DeckSection.Commander.validate(card))
```

Comment:
I edited `DeckSection.matchingSection` to make it more generally usable, by removing the test that put all legendary creatures in the commander section. While investigating to see if this would break any current uses of it, I stumbled upon DeckRecognizerTest.java and ran it. There was a corner case test called `testCornerCaseWhenThereIsNoCurrentSectionAndMatchedSectionIsNotAllowedButCouldMatchMain()` (might be the longest symbol name I've ever seen by the way), where it was apparently the intended behavior that `DeckSection.matchingSection` would return `DeckSection.Commander` for commander candidates.

That was the only case I could find where it seemed to matter, and since there was a test for it specifically, I assumed it was intentional. So I shifted the responsibility of choosing the Commander section outward to that `getTokenSection` method, in the form of the lines there.